test/src/jest-xpaths.js

```diff
@@ -0,0 +1,19 @@
+import 'expect-puppeteer';
+
+// Some extensions to the Jest expect API to better support XPaths
+expect.extend({
+  async toContainXPath(page, xpath) {
+    return page.waitForXPath(xpath)
+      .then(() => ({ pass: true }));
```

Comment:
Is there a reason you're not using `await` here, and just returning an object afterwards?

Reply:
I was thinking `then/catch` would make for more elegant exception handling in the case where the XPath wasn't found, but I forgot to loop back and implement it. I tried it and a normal try/catch seems cleaner to me.